library/src/main/java/com/novoda/downloadmanager/DownloadManager.java

```diff
@@ -97,23 +99,24 @@ public void pause(DownloadBatchId downloadBatchId) {
     public void resume(DownloadBatchId downloadBatchId) {
         DownloadBatch downloadBatch = downloadBatchMap.get(downloadBatchId);
         if (downloadBatch == null) {
+            Log.v("abort resume batch " + downloadBatchId + " will not be resume as it does not exists in the running batches map");
             return;
         }
 
         if (downloadBatch.status().status() == DownloadBatchStatus.Status.DOWNLOADING) {
+            Log.v("abort resume batch " + downloadBatchId + " will not be resume as it's already downloading");
             return;
         }
 
-        downloadBatchMap.remove(downloadBatchId);
```

Comment:
this is not needed and makes it complex to understand. I've added a check in the actual `downloader.download(..)` method for adding it into the map if is necessary

Reply:
> method for adding it into the map if is necessary.
but here you are removing it 😄